QWeb/internal/browser/__init__.py

```diff
@@ -14,82 +14,82 @@
-    "font/ttf;"
-    "font/woff2;"
-    "font/woff;"
-    "image/gif;"
-    "image/jpeg;"
-    "image/png;"
-    "image/svg+xml;"
-    "image/tiff;"
-    "image/webp;"
-    "image/x-icon;"
-    "text/calendar;"
-    "text/css;"
-    "text/csv;"
-    "text/html;"
-    "text/plain;"
-    "video/3gpp2audio/3gpp2;"
-    "video/3gppaudio/3gpp;"
-    "video/mpeg;"
-    "video/ogg;"
-    "video/webm;"
-    "video/x-msvideo")
-
-
-_current_browser = None
-_open_browsers = []
-
-
-def get_current_browser():
+MIME_TYPES = ("application/epub+zip;"
+              "application/java-archive;"
+              "application/javascript;"
+              "application/json;"
+              "application/msword;"
+              "application/octet-stream;"
+              "application/octet-stream;"
+              "application/ogg;"
+              "application/pdf;"
+              "application/rtf;"
+              "application/typescript;"
+              "application/vnd.amazon.ebook;"
+              "application/vnd.apple.installer+xml;"
+              "application/vnd.mozilla.xul+xml;"
+              "application/vnd.ms-excel;"
+              "application/vnd.ms-fontobject;"
+              "application/vnd.ms-powerpoint;"
+              "application/vnd.oasis.opendocument.text;"
+              "application/vnd.oasis.opendocument;"
+              "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;"
+              "application/vnd.visio;"
+              "application/x-7z-compressed"
+              "application/x-abiword;"
+              "application/x-bzip2;"
+              "application/x-bzip;"
+              "application/x-csh;"
+              "application/x-rar-compressed;"
+              "application/x-sh;"
+              "application/x-shockwave-flash;"
+              "application/x-tar;"
+              "application/xhtml+xml;"
+              "application/xml;"
+              "application/zip;"
+              "audio/aac;"
+              "audio/midi;"
+              "audio/ogg;"
+              "audio/webm;"
+              "audio/x-wav;"
+              "font/otf;"
+              "font/ttf;"
+              "font/woff2;"
+              "font/woff;"
+              "image/gif;"
+              "image/jpeg;"
+              "image/png;"
+              "image/svg+xml;"
+              "image/tiff;"
+              "image/webp;"
+              "image/x-icon;"
+              "text/calendar;"
+              "text/css;"
+              "text/csv;"
+              "text/html;"
+              "text/plain;"
+              "video/3gpp2audio/3gpp2;"
+              "video/3gppaudio/3gpp;"
+              "video/mpeg;"
+              "video/ogg;"
+              "video/webm;"
+              "video/x-msvideo")
+
+_current_browser: WebDriver = None  # type: ignore[assignment]
+_open_browsers: list[WebDriver] = []
+
+
+def get_current_browser() -> WebDriver:
     return _current_browser
 
 
-def set_current_browser(index):
+def set_current_browser(index: str) -> None:
```

Comment:
Can't this be int as well? Or was it so that it always comes from robot script as string?

Reply:
True that it could also be int.
Problem is that the first check of the function, `if index.isdigit():`, causes an AttributeError if an int is passed as a parameter

This function was only called from `keywords/browser.py - switch_browser` where the parameter is cast to str

Refactored the functions to allow passing param `index` as an int
I changed the type to `Union[int, str]` and removed the str cast from the `switch_browser` kw.

Commit
[d18ce3f](https://github.com/qentinelqi/qweb/pull/76/commits/d18ce3f625e6568b5da423a830f0fa9bc6a0281d)